test/cpp/ext/otel/otel_tracing_test.cc

```diff
@@ -0,0 +1,545 @@
+      [](const SpanDataEvent& event) {
+        return event.GetName() == "Outbound message compressed";
+      });
+  EXPECT_NE(outbound_message_compressed_event,
+            (*server_span)->GetEvents().end());
+  EXPECT_THAT(
+      outbound_message_compressed_event->GetAttributes(),
+      UnorderedElementsAre(
+          Pair("sequence-number", VariantWith<uint64_t>(0)),
+          Pair("message-size-compressed", VariantWith<uint64_t>(Lt(31)))));
+}
+
+TEST_F(OTelTracingTest, FailedStatus) {
+  {
+    EchoRequest request;
+    request.set_message("foo");
+    request.mutable_param()->mutable_expected_error()->set_code(
+        grpc::StatusCode::UNAVAILABLE);
+    request.mutable_param()->mutable_expected_error()->set_error_message(
+        "test message");
+    EchoResponse response;
+    grpc::ClientContext context;
+    context.set_compression_algorithm(GRPC_COMPRESS_GZIP);
+    grpc::Status status = stub_->Echo(&context, request, &response);
+  }
+  absl::SleepFor(absl::Milliseconds(500 * grpc_test_slowdown_factor()));
+  auto spans = data_->GetSpans();
+  EXPECT_EQ(spans.size(), 3);
+  const auto attempt_span = std::find_if(
+      spans.begin(), spans.end(), [](const std::unique_ptr<SpanData>& span) {
+        return span->GetName() == "Attempt.grpc.testing.EchoTestService/Echo";
+      });
+  EXPECT_NE(attempt_span, spans.end());
+  EXPECT_EQ((*attempt_span)->GetStatus(),
+            opentelemetry::trace::StatusCode::kError);
+  EXPECT_THAT((*attempt_span)->GetDescription(),
+              MatchesRegex("UNAVAILABLE:.*test message.*"));
+  const auto server_span = std::find_if(
+      spans.begin(), spans.end(), [](const std::unique_ptr<SpanData>& span) {
+        return span->GetName() == "Recv.grpc.testing.EchoTestService/Echo";
+      });
+  EXPECT_NE(server_span, spans.end());
+  EXPECT_EQ((*server_span)->GetStatus(),
+            opentelemetry::trace::StatusCode::kError);
+  EXPECT_THAT((*server_span)->GetDescription(),
+              MatchesRegex("UNAVAILABLE:.*test message.*"));
+}
+
+TEST_F(OTelTracingTest, Streaming) {
+  {
+    EchoRequest request;
+    request.set_message("foo");
+    EchoResponse response;
+    grpc::ClientContext context;
+    auto stream = stub_->BidiStream(&context);
+    for (int i = 0; i < 10; ++i) {
+      EXPECT_TRUE(stream->Write(request));
+      EXPECT_TRUE(stream->Read(&response));
+    }
+    stream->WritesDone();
+    EXPECT_TRUE(stream->Finish().ok());
+  }
+  absl::SleepFor(absl::Milliseconds(500 * grpc_test_slowdown_factor()));
+  auto spans = data_->GetSpans();
+  EXPECT_EQ(spans.size(), 3);
+  const auto attempt_span = std::find_if(
+      spans.begin(), spans.end(), [](const std::unique_ptr<SpanData>& span) {
+        return span->GetName() ==
+               "Attempt.grpc.testing.EchoTestService/BidiStream";
+      });
+  EXPECT_NE(attempt_span, spans.end());
+  // Verify messages on the attempt span
+  std::vector<uint64_t> outbound_seq_nums;
```

Comment:
Suggest using `std::set<>` here, so that the elements are ordered.  That way, we can use `ElementsAre()` instead of `UnorderedElementsAre()` below.

Reply:
I don't see the reason to use an std::set. We can use `ElementsAre()` irrespective.